### DISCUSSION TITLE
Inventree Javadocs

### DISCUSSION THREAD

**Discussion by CodingPupper3033** [inventree/InvenTree#5225] [Q&A]
Hello again,

I see than many of the functions exposed to plugins, for plugin use, are python docstringed.

Excuse my Or*cle, but is there a python Inventree version of Javadocs anywhere for me to read? Something like (this)[https://docs.oracle.com/javase/7/docs/api/java/lang/String.html]. It would be easier to read it in a page format without any code. I really haven't played with pythons' version of Javadocs, so have no clue how to access them.

In addition, is there a list of everything exposed to plugins? The metadata section on plugins shows how to access Parts, not stock, but stock is there, and I believe able to be used. The mixins show some of their functions, but I don't believe all of them. Also, I would imagine there are more tools exposed that I am not even aware of. How would I know what tools are in my toolbox?

Also, I really appreciate you guys spending the time answering all my noobie questions, your support means a lot to me!

**Comment by CodingPupper3033** [comment]
To answer the last part of my question, if I'm not mistaken, [this](https://github.com/inventree/inventree-python/tree/master) is a link to the python library for plugins to use. However, I will continue to look for an answer to my other questions.

**Comment by wolflu05** [comment]
Hello, docs are currently not the first priority and nobody picked this up. But there is a feature request to add some auto generated docs to the docs page by using python docstrings. #4702

The python library you linked is for usage from an external script that runs e.g. on your computer. This library uses the HTTP api, for plugins using the models directly is the recommended approach.

**Comment by matmair** [comment]
Shoe answer: The code is the docs.
A bit longer: We are a small team and fast moving so docs are... difficult, especially for the more niche parts. Newer code is pretty well documented with docstrings, old stuff not so much. Regarding plugins we have pages with short introductions for all mixins (eg [ActionMixin](https://docs.inventree.org/en/latest/extend/plugins/action/)). If you run into something undocumented feel free to raise a documentation issue or create a PR, some day we will get to it.
